.github/workflows/ci.yml

```diff
@@ -0,0 +1,45 @@
+name: Linting and Tests
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+
+jobs:
+  #check-code-style:
+    #name: Check code style
+    #runs-on: ubuntu-latest
+    #steps:
+      #- uses: actions/checkout@v2
+      #- uses: actions/setup-python@v2
+        #with:
+          #python-version: "3.x"
+      #- run: python -m pip install tox
+      #- run: tox -e check_codestyle
+
+  #check-types:
+    #name: Check types with Mypy
+    #runs-on: ubuntu-latest
+    #steps:
+      #- uses: actions/checkout@v2
+      #- uses: actions/setup-python@v2
+        #with:
+          #python-version: "3.x"
+      #- run: python -m pip install tox
+      #- run: tox -e check_types
+
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Run the unit tests both against our oldest supported Python version
+        # and the newest stable.
+        python_version: [ "3.7", "3.x" ]
```

Comment:
I have no idea what we currently support for this, but should we test 3.7 / 3.8 / 3.9 / 3.10?

Reply:
No objection to doing so, but for some reason we decided to do oldest+newest for modules. Was it perhaps because of the burden of updating the CI files for each release?

Reply:
(and current CI does Py 3.5 but I see no reason to support 3.5, 3.6 when Synapse itself isn't supported on those)

Reply:
That's fine! Wasn't sure if this was straight from the template or if you had modified it.